@xen-orchestra/backups/runBackupWorker.js

```diff
@@ -0,0 +1,39 @@
+const path = require('path')
+const pDefer = require('promise-toolbox/defer')
+const { createLogger } = require('@xen-orchestra/log')
+const { fork } = require('child_process')
+
+const { warn } = createLogger('xo:backups:backupWorker')
+
+exports.runBackupWorker = function runBackupWorker(params, onLog) {
+  const { promise, resolve, reject } = pDefer()
+
+  const worker = fork(path.resolve(__dirname, '_backupWorker.js'))
+
+  worker.on('exit', code => reject(new Error(`worker exited with code ${code}`)))
+  worker.on('error', reject)
+
+  worker.on('message', ({ type, data, status, result }) => {
+    try {
+      if (type === 'result') {
+        if (status === 'success') {
+          resolve(result)
+        } else {
+          reject(result)
+        }
+      } else {
+        onLog(data)
```

Comment:
Test or assert the type here, so that new messages will not break existing flow.

Reply:
Normally will not implement a half of new functionalities. So, i don't think it's necessary.

Reply:
IMHO, we shouldn't  do an implementation for a need and in the same time responding to a future need which can probably not been implemented.